net/net.go

```diff
@@ -642,7 +642,10 @@ func (n *net) AddRecord(ctx context.Context, id thread.ID, lid peer.ID, rec core
 	if err = rec.Verify(logpk); err != nil {
 		return err
 	}
-	return n.PutRecord(ctx, id, lid, rec)
+	if err = n.PutRecord(ctx, id, lid, rec); err != nil {
+		return err
+	}
+	return n.server.pushRecord(ctx, id, lid, rec)
```

Comment:
Surely this must have been a bug... why not push out records from `AddRecord` as well as `CreateRecord`?

Reply:
Yeah. Eventually, this can be different if we plan to batch pushes or similar, but not the case now.
SGTM.

Reply:
Nice yeah, good call